lib/chai/core/assertions.js

```diff
@@ -1500,6 +1505,85 @@ module.exports = function (chai, _) {
   Assertion.addMethod('keys', assertKeys);
   Assertion.addMethod('key', assertKeys);
 
+  function assertStrings (strings) {
+    _.expectTypes(this, ['string']);
+    var obj = flag(this, 'object')
+      , negate = flag(this, 'negate')
+      , stringsType = _.type(strings)
+      , str
+      , deepStr = ''
+      , ok = true
+      , mixedArgsMsg = 'when testing strings against strings you must give a single Array|String argument or multiple String arguments';
+
+    if (stringsType === 'Array') {
+      if (arguments.length > 1) throw new Error(mixedArgsMsg);
+    } else {
+      strings = Array.prototype.slice.call(arguments)
+    }
+
+    if (strings.length === 1) {
+      return this.string(strings[0])
+    }
+
+    if (!strings.length) throw new Error('strings required');
+
+    var len = strings.length
+      , any = flag(this, 'any')
+      , all = flag(this, 'all')
+      , expected = strings;
+
+    if (!any && !all) {
+      all = true;
+    }
+
+    function containsStr (expectedStr) {
+      return ~obj.indexOf(expectedStr)
+    }
+
+    // Has any
+    if (any) {
+      var reducer = negate ? 'every' : 'some'
+      ok = expected[reducer](containsStr)
+    }
+
+    // Has all
+    if (all) {
+      var reducer = negate ? 'some' : 'every'
```

Comment:
`reducer` is already declared here, because the `if (any)` above.
I suggest you to move the declaration of this `reducer` variable to before both ifs